### PR TITLE
Put peqp in a separate volume called "static"

### PR DESCRIPTION
**Pull request by icyflame**
I hope this is what you meant in [your comment](https://github.com/metakgp/metakgp-wiki/issues/6#issuecomment-328897499) in #6 . The changes were pretty easy to make once I got the hang of it. :smile: 

- This patch makes the `mediawiki` volume completely ephemeral
- All static assets can be served from this subdomain: `static.metakgp.org`
- Serving static assets from the same domain as the wiki turns out to be tricky, so I avoided it. It's just a DNS record and we can easily keep track of it.
- The script at the end of this patch does the same thing as the restore mysql DB does. It is convenient to use:
    - On a pc in the KGP network: `wget -r 10.17.32.9`
    - Compress: `tar czf peqp.tar.gz 10.17.32.9/`
    - Rsync to server: `rsync -avzhe "ssh" peqp.tar.gz root@IP:~`
    - Extract at server: `tar zxf peqp.tar.gz`
    - Move to `metakgp-wiki` and then call the script: `./script/restore-peqp-to-static.sh 10.17.32.9` (The first argument is the directory that consists the peqp folder alongwith all the PDFs)

    I will add this to the runbook as soon as these two patches are merged in after testing it one more time.

### CODE REVIEW

File: nginx/Dockerfile
Comment (amrav) [inline nginx/Dockerfile]: Can you rename these to wiki.metakgp.org and static.metakgp.org?

File: nginx/metakgp.org
Comment (amrav) [inline nginx/metakgp.org:30]: Unfortunately I _think_ this causes things like locahost to stop working, which we want for development. Can you verify whether this is the case, and see if there's a way around it?

File: nginx/metakgp.org
Comment (amrav) [inline nginx/metakgp.org:27]: For the mediawiki volume to be truly ephemeral, we also want to put image uploads into the static volume. You could take a look at whether it's possible to configure media uploads to a separate subdomain, but I think the simplest solution might be to symlink the images dir to a dir in the static volume. Feel free to defer this to a later patch though.

File: nginx/static
Comment (amrav) [inline nginx/static]: Does this turn off showing an index of all the files? I think we want that to be off, but what do you think?

File: nginx/static
Comment (amrav) [inline nginx/static]: We can get rid of this boilerplate, I think.

File: docker-compose.override.yml
Comment (amrav) [inline docker-compose.override.yml]: I think this belongs to the prod configuration - default and override should only contain config for development.

File: scripts/restore-peqp-to-static.sh
Comment (amrav) [inline scripts/restore-peqp-to-static.sh:7]: I have a slight preference against having scripts to do very simple things, like run `docker cp`. Perhaps this belongs to the runbook instead?

**Review by amrav — CHANGES_REQUESTED** [review]
Looks good, almost ready to merge - thanks!

File: nginx/static
Comment (icyflame) [reply nginx/static]: Yeah, I think we should turn it off too. The default is `off`. I will keep this in the config to keep it explicit.

File: scripts/restore-peqp-to-static.sh
Comment (icyflame) [reply scripts/restore-peqp-to-static.sh:7]: Hmm, We could put this in the runbook, this script just felt natural beside the mysql backup restore script. I think we should keep it.

File: nginx/metakgp.org
Comment (icyflame) [reply nginx/metakgp.org:27]: Yeah, I got it. Will defer to a later patch.

File: nginx/metakgp.org
Comment (icyflame) [reply nginx/metakgp.org:30]: Hey, I don't have docker installed on my computer. I did `wget localhost` on the server and that just gave me a 301 to the server URL. I will see if I have the space on my computer to set the whole thing up and check this.

**Comment by icyflame** [comment]
Rebased to latest master.

Remaining:

- [x] Check whether `localhost` works
- [x] Decide whether to have a separate script for restoring PEQP or put that in the runbook (I prefer having a script)

**Comment by icyflame** [comment]
image

@amrav Localhost still works. I just setup a new server and tried it and it works. I have rebased to latest master and used this particular build to set this up. Please review and approve! :slightly_smiling_face:

**Comment by icyflame** [comment]
Hey @amrav, I have solved the issues now. Can you please review and merge? I think all the requirements have been met.

**Review by amrav — APPROVED** [review]

**Comment by amrav** [comment]
Lgtm if it works, but do you understand why nginx serves the correct domain for localhost? Why isn't it defaulting to static.metakgp.org, for instance?

**Comment by icyflame** [comment]
@amrav Okay, the mechanism was easy to understand. It's well documented [here](https://nginx.org/en/docs/http/request_processing.html)

In particular:

image

A verbose log of the request:

image

So, the `Host: localhost` doesn't match anything configured and it just routes the request to the `default_server` which we have configured to be the wiki server.

I hope that clears it up. This was a good thing to find out! :smile: Okay to merge this patch now?

**Comment by amrav** [comment]
Sounds good, thanks for clarifying!

**Pull request merged**
